src/commonMain/kotlin/fr/acinq/lightning/channel/Normal.kt

```diff
@@ -627,17 +643,17 @@ data class Normal(
                         } ?: logger.warning { "cannot find payment for $htlc" }
                     }
                 }
-                // if we are splicing and are earlyè in the process, then we cancel it
+                // If we are splicing and are early in the process, then we cancel it.
+                val replyTo = when (spliceStatus) {
+                    is SpliceStatus.Requested -> spliceStatus.command.replyTo
+                    is SpliceStatus.InProgress -> spliceStatus.replyTo
+                    else -> null
+                }
+                replyTo?.complete(Command.Splice.Response.Failure.Disconnected)
+                // If we didn't reach the signatures exchange step, we implicitly abort the splice.
```

Comment:
I don't think this refactoring makes things clearer, previous version made it obvious that it's either replyTo+forget splice, or keep going. If we keep this, maybe rename `replyTo` to `spliceReplyTo`?

Reply:
I re-grouped them in https://github.com/ACINQ/lightning-kmp/pull/438/commits/9b9810ffc67da8f0b40825b2126d1dc54ee6b0dd
I think it's important to explicitly list every case, as `SpliceStatus.Aborted` must be converted to `SpliceStatus.None` on disconnection.